mautic/mautic
# Add support to check duplicates for api/companies/batch/new

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description: 

This PR enabled check unique identified fields for company API create/ and batch create endpoint.




#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create batch API  with any combination of unique fields

`POST api/companies/batch/new`


```
[
    {
        "companyname": "test",
        "custom_company_field": "x2"
    },
    {
        "companyname": "test2",
        "custom_company_field": "x1"
    }
]
```
3. Run it again. It should not create new companies, because API now detect duplicate companies
